src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -51,6 +52,38 @@ def str_to_sql(value: str) -> str:
     return f"'{sql_str}'"
 
 
+def str_to_sql_for_year_month_interval(
+    value: str, datatype: YearMonthIntervalType
+) -> str:
+    """
+    Converts "INTERVAL YY-MM [YEAR TO MONTH | YEAR | MONTH]" to quoted format, optimizing based on datatype:
+    - Same start/end field: extracts specific value (YEAR or MONTH only)
+    - Different fields: uses full "YEAR TO MONTH" format
+
+    Examples:
+        "INTERVAL 1-2 YEAR TO MONTH" -> "INTERVAL '1-2' YEAR TO MONTH"
+        "INTERVAL 5-0 YEAR TO MONTH" + YearMonthIntervalType(0,0) -> "INTERVAL '5' YEAR"
+        "INTERVAL 0-3 YEAR TO MONTH" + YearMonthIntervalType(1,1) -> "INTERVAL '3' MONTH"
```

Comment:
another clarifying question: we strictly do not support input value `INTERVAL '1-2' YEAR TO MONTH` right? There must be no quotes around YY-MM

Reply:
That's not what I meant to ask. let me rephrase. 

Currently `str_to_sql_for_year_month_interval("INTERVAL '1-11' YEAR TO MONTH", datatype)` is invalid. This function expects no quotes around YY-MM. Is that the desired behavior?